backend/adapter_processor/models.py

```diff
@@ -93,6 +98,28 @@ class Meta:
             ),
         ]
 
+    def get_adapter_meta_data(self) -> Any:
+        encryption_secret: str = settings.ENCRYPTION_KEY
+        f: Fernet = Fernet(encryption_secret.encode("utf-8"))
+
+        adapter_metadata = json.loads(
+            f.decrypt(bytes(self.adapter_metadata_b).decode("utf-8"))
+        )
+        return adapter_metadata
```

Comment:
@johnyrahul Can we create a generic function for get_metadata. Seems the similar code we created some other places to get other metadata (ex: connector).

Reply:
@muhammad-ali-e  yes, we can use this method in all remaining places to get the metadata. This method is tied with the instance itself.